clients/src/main/java/org/apache/kafka/clients/Metadata.java

```diff
@@ -127,6 +127,15 @@ public synchronized void addTopics(String... topics) {
     }
 
     /**
+     * Add one or more topics to maintain metadata for
+     */
+    public synchronized void addTopics(Iterable<String> topics) {
+        for (String topic : topics)
+            this.topics.add(topic);
+        requestUpdate();
```

Comment:
Yep, sounds reasonable. In fact, even better would be to check if the metadata already includes all of the provided topics and only then use `requestUpdate()`.